src/toil/jobStores/abstractJobStore.py

```diff
@@ -1449,6 +1450,71 @@ def read_logs(self, callback: Callable[..., Any], read_all: bool = False) -> int
         """
         raise NotImplementedError()
 
+    # A few shared files useful to Toil, but probably less useful to the users.
+
+    def write_leader_pid_file(self) -> None:
+        """
+        Write the pid of this process to a file in the job store.
+
+        Overwriting the current contents of pid.log is a feature, not a bug of
+        this method. Other methods will rely on always having the most current
+        pid available. So far there is no reason to store any old pids.
+        """
+        with self.write_shared_file_stream('pid.log') as f:
+            f.write(str(os.getpid()).encode('utf-8'))
+
+    def read_leader_pid_file(self) -> int:
+        """
+        Read the pid of the leader process to a file in the job store.
+
+        :raise NoSuchFileException: If the PID file doesn't exist.
+        """
+        with self.read_shared_file_stream('pid.log') as f:
+            return int(f.read().strip())
+
+    def write_leader_node_id(self) -> None:
```

Comment:
The names are a bit weird; the PID file is a `_file`, but this is just an `_id`. Maybe it should also have a `_file` suffix? Or maybe we should drop it from the PID functions so we get `write_leader_pid` and `write_leader_node_id`?